Allow download and upload of Kura native snapshot

lorthirk opened the pull request:
Currently, Device Snapshots downloaded from the Console or REST APIs cannot be directly applied to a Kura device, since the XML that is downloaded from the Console has a different format the one that Kura can deserialize. With this PR, bot "Standard" (old format) and "Native" (new, Kura-compatible format) snapshots can be both dowloaded and uploaded from the console. Native snapshots can also be directly applied to a device from the Kura console.

**Related Issue**
Fixes #2704

**Description of the solution adopted**
Translation logic has been extracted from `TranslatorAppConfigurationKuraKapua` and `TranslatorAppConfigurationKapuaKura` to a utility class named `KuraDeviceConfigurationUtils`, in the `kapua-device-call-kura` module. The same logic is also called from the Download and Upload servlets to do necessary conversions.

**Screenshots**
![Schermata 2020-02-20 alle 17 08 19](https://user-images.githubusercontent.com/8140139/74954645-b7974280-5403-11ea-989e-81589061fd95.png)
![Schermata 2020-02-20 alle 17 08 22](https://user-images.githubusercontent.com/8140139/74954650-b8c86f80-5403-11ea-88b3-9f6923f66d8d.png)


**Any side note on the changes made**
Do we agree on "Native" and "Standard" names?

Commit:
Allow download and upload of Kura native snapshot

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

Coduz reviewed (changes requested):
"Standard" term alternative wanted.

💰 1.000.000K reward 💰

LeoNerdoG commented:
> "Standard" term alternative wanted.
> 
> 💰 1.000.000K reward 💰

I suggest "accepted", "common" or "general". 
Did I get the prize? 😸 😸

kcsj17 commented:
Can this be used to upload, via Kapua, a wiregraph that was downloaded via Kura web interface?

LeoNerdoG commented:
Hi @kcsj17,
yes, you can either upload just a wire snapshot or snapshot of the whole system. 
Just be sure that you select the "Native" option in the UI as @lorthirk described above. 

I have just tested this and it works fine.

lorthirk commented:
After discussing with @Coduz and @stefanomorson, the following changes have been proposed:

- The format names will be "Platform" and "Device"
- Only one "Upload and Apply" button that can support and autodetect both formats
- The REST API has to be enhanced with the same support (autodetect for upload, query string parameter to choose download format)

Commit:
Address PR comments
- Format names renamed to "Platform" and "Device"
- "Upload and Apply" button now supports both formats
- Double format REST API implementation

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

Coduz commented:
Removing project tracking since it is tracked by the issue

lorthirk commented:
Still needs some thinking as it involves other parts that were overlooked at the beginning. Adding the `proposal` label
